File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>
using namespace std;

bool solution(int x) {
    bool answer = true;
    string num = to_string(x);
    int sum = 0;
    for(auto n : num)
    {
        sum += n-'0';
    }
    if(x%sum == 0) answer = true;
    else answer = false;
    return answer;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
using System;
using System.Linq;
public class Solution {
    public bool solution(int x) {
        string num = x.ToString();
        int[] arr = num.Select(a => int.Parse(a.ToString())).ToArray();
        int sum = arr.Sum();
        if(x % sum == 0) return true;
        else return false;
    }
}